pandas/tests/io/json/test_pandas.py

```diff
@@ -154,322 +154,212 @@ def test_frame_non_unique_columns_raises(self, orient):
+    @pytest.mark.parametrize("convert_axes", [True, False])
+    @pytest.mark.parametrize("numpy", [True, False])
+    def test_roundtrip_timestamp(self, orient, convert_axes, numpy):
+        # TODO: improve coverage with date_format parameter
+        data = self.tsframe.to_json(orient=orient)
+        result = pd.read_json(
+            data, orient=orient, convert_axes=convert_axes, numpy=numpy
         )
+        expected = self.tsframe.copy()
+
+        if not convert_axes:  # one off for ts handling
+            # DTI gets converted to epoch values
+            idx = expected.index.astype(np.int64) // 1000000
+            if orient != "split":  # TODO: handle consistently across orients
+                idx = idx.astype(str)
+
+            expected.index = idx
 
-        # time series data
-        _check_all_orients(self.tsframe)
+        if orient == "records" or orient == "values":
+            expected = expected.reset_index(drop=True)
+        if orient == "values":
+            expected.columns = range(len(expected.columns))
+
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("convert_axes", [True, False])
+    @pytest.mark.parametrize("numpy", [True, False])
+    def test_roundtrip_mixed(self, orient, convert_axes, numpy):
+        if numpy and orient != "split":
+            pytest.xfail("Can't decode directly to array")
 
-        # mixed data
         index = pd.Index(["a", "b", "c", "d", "e"])
-        data = {
+        values = {
             "A": [0.0, 1.0, 2.0, 3.0, 4.0],
             "B": [0.0, 1.0, 0.0, 1.0, 0.0],
             "C": ["foo1", "foo2", "foo3", "foo4", "foo5"],
             "D": [True, False, True, False, True],
         }
-        df = DataFrame(data=data, index=index)
-        _check_orient(df, "split", check_dtype=False)
-        _check_orient(df, "records", check_dtype=False)
-        _check_orient(df, "values", check_dtype=False)
-        _check_orient(df, "columns", check_dtype=False)
-        # index oriented is problematic as it is read back in in a transposed
-        # state, so the columns are interpreted as having mixed data and
-        # given object dtypes.
-        # force everything to have object dtype beforehand
-        _check_orient(df.transpose().transpose(), "index", dtype=False)
+
+        df = DataFrame(data=values, index=index)
+
+        data = df.to_json(orient=orient)
+        result = pd.read_json(
+            data, orient=orient, convert_axes=convert_axes, numpy=numpy
+        )
+
+        expected = df.copy()
+        expected = expected.assign(**expected.select_dtypes("number").astype(np.int64))
+
+        if not numpy and (orient == "index" or (PY35 and orient == "columns")):
+            expected = expected.sort_index()
+
+        if orient == "records" or orient == "values":
+            expected = expected.reset_index(drop=True)
+        if orient == "values":
+            expected.columns = range(len(expected.columns))
```

Comment:
re-iterating request to share code for ~344-360.  As is I have to look at each version for "is this subtly different and if so why"

Reply:
Hmm I don't know I fully understand what you are asking for. This particular test lines up pretty well to the left (lines 463 - 472) but is more explicit about the expected shape of the result. This idiom of resetting particular axes for values and records is used throughout the module (probably worth a dedicated function, but leaving to a follow up)

Reply:
>  This idiom of resetting particular axes for values and records is used throughout the module

Yah, thats what I'm asking for.  OK for follow-up.